### PR TITLE
MOTECH-2838 Unit tests

### PR DESCRIPTION
**Pull request by filgomul**
Unit tests for classes added in #383

### CODE REVIEW

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

File: openmrs/src/test/java/org/motechproject/openmrs/tasks/impl/OpenMRSActionProxyServiceTest.java [hunk 1]
```diff
@@ -377,6 +398,37 @@ public void shouldChangeStateOfProgramEnrollment() {
...
+        String cohortQueryUuid = "QQQ";
+
+        ArgumentCaptor<String> uuidCaptor = ArgumentCaptor.forClass(String.class);
+        ArgumentCaptor<Map> parametersCaptor = ArgumentCaptor.forClass(Map.class);
+
+        openMRSActionProxyService.getCohortQueryReport(CONFIG_NAME, cohortQueryUuid, null);
+
+        verify(cohortService).getCohortQueryReport(eq(CONFIG_NAME), uuidCaptor.capture(), parametersCaptor.capture());
+        assertEquals(cohortQueryUuid, uuidCaptor.getValue());
+        assertEquals(null, parametersCaptor.getValue());
```
Comment (kmadej) [inline openmrs/src/test/java/org/motechproject/openmrs/tasks/impl/OpenMRSActionProxyServiceTest.java]: assertNull

**Comment by martokarski** [comment]
test this please

File: openmrs/src/test/java/org/motechproject/openmrs/resource/impl/CohortResourceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        URI url = config.toInstancePathWithParams("/reportingrest/cohort/{uuid}", cohortQueryReport.getUuid());
+
+        when(restOperations.exchange(eq(url), eq(HttpMethod.GET), any(HttpEntity.class), eq(String.class)))
+                .thenReturn(getResponseFromFile(GET_COHORT_QUERY_REPORT_RESPONSE_JSON));
+
+        CohortQueryReport created = cohortResource.getCohortQueryReport(config, cohortQueryReport.getUuid(), null);
+
+        verify(restOperations).exchange(eq(url), eq(HttpMethod.GET), requestCaptor.capture(), eq(String.class));
+
+        assertThat(created, equalTo(cohortQueryReport));
```
Comment (kmadej) [inline openmrs/src/test/java/org/motechproject/openmrs/resource/impl/CohortResourceImplTest.java]: expected value should be on the left side

File: openmrs/src/test/java/org/motechproject/openmrs/resource/impl/CohortResourceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        doReturn(getResponseFromFile(GET_COHORT_QUERY_REPORT_RESPONSE_JSON))
+                .when(restOperations).exchange(eq(url), eq(HttpMethod.GET), any(HttpEntity.class), eq(String.class));
+
+        CohortQueryReport created = cohortResource.getCohortQueryReport(config, cohortQueryReport.getUuid(), parameters);
+
+        verify(restOperations).exchange(eq(url), eq(HttpMethod.GET), requestCaptor.capture(), eq(String.class));
+
+        assertThat(created, equalTo(cohortQueryReport));
+        assertThat(requestCaptor.getValue().getHeaders(), equalTo(getHeadersForGet(config)));
+        assertThat(JsonUtils.readJson(requestCaptor.getValue().getBody(), JsonObject.class), nullValue());
```
Comment (kmadej) [inline openmrs/src/test/java/org/motechproject/openmrs/resource/impl/CohortResourceImplTest.java]: why do you assert this json with null ?

File: openmrs/src/test/java/org/motechproject/openmrs/tasks/impl/OpenMRSActionProxyServiceTest.java [hunk 4]
```diff
@@ -375,6 +408,63 @@ public void shouldChangeStateOfProgramEnrollment() {
...
+    public void shouldGetEmptyCohortQueryReport() {
+        String cohortQueryUuid = "QQQ";
+
+        ArgumentCaptor<String> uuidCaptor = ArgumentCaptor.forClass(String.class);
+        ArgumentCaptor<Map> parametersCaptor = ArgumentCaptor.forClass(Map.class);
+
+        doReturn(prepareEmptyCohortQueryReport()).when(cohortService)
+                .getCohortQueryReport(eq(CONFIG_NAME), eq(cohortQueryUuid), eq(null));
+
+        openMRSActionProxyService.getCohortQueryReport(CONFIG_NAME, cohortQueryUuid, null);
```
Comment (kmadej) [inline openmrs/src/test/java/org/motechproject/openmrs/tasks/impl/OpenMRSActionProxyServiceTest.java]: why do you send null here ?

File: openmrs/src/test/java/org/motechproject/openmrs/tasks/impl/OpenMRSActionProxyServiceTest.java [hunk 5]
```diff
@@ -438,4 +528,26 @@ private Concept createTestConcept(String testConceptName) {
...
+        cohortQueryReport.setMembers(new ArrayList<>());
+
+        return cohortQueryReport;
+    }
+
+    private CohortQueryReport prepareCohortQueryReportWithMembers() {
+        CohortQueryReport cohortQueryReport = new CohortQueryReport();
+
+        List<CohortQueryReportMember> members = Arrays.asList(
+                new CohortQueryReportMember[]{
```
Comment (kmadej) [inline openmrs/src/test/java/org/motechproject/openmrs/tasks/impl/OpenMRSActionProxyServiceTest.java]: redundant, you do not have to create array

File: openmrs/src/test/java/org/motechproject/openmrs/tasks/impl/OpenMRSActionProxyServiceTest.java [hunk 6]
```diff
@@ -375,6 +408,63 @@ public void shouldChangeStateOfProgramEnrollment() {
...
+        ArgumentCaptor<String> uuidCaptor = ArgumentCaptor.forClass(String.class);
+        ArgumentCaptor<Map> parametersCaptor = ArgumentCaptor.forClass(Map.class);
+
+        doReturn(prepareCohortQueryReportWithMembers()).when(cohortService)
+                .getCohortQueryReport(eq(CONFIG_NAME), eq(cohortQueryUuid), eq(parameters));
+
+        openMRSActionProxyService.getCohortQueryReport(CONFIG_NAME, cohortQueryUuid, parameters);
+
+        verify(cohortService).getCohortQueryReport(eq(CONFIG_NAME), uuidCaptor.capture(), parametersCaptor.capture());
+        verify(eventRelay, times(3)).sendEventMessage(Matchers.<MotechEvent>any());
```
Comment (kmadej) [inline openmrs/src/test/java/org/motechproject/openmrs/tasks/impl/OpenMRSActionProxyServiceTest.java]: you should verify if appropriate parameters are sent in motech event

File: openmrs/src/test/java/org/motechproject/openmrs/tasks/impl/OpenMRSActionProxyServiceTest.java [hunk 7]
```diff
@@ -375,6 +408,63 @@ public void shouldChangeStateOfProgramEnrollment() {
...
+
+        doReturn(prepareCohortQueryReportWithMembers()).when(cohortService)
+                .getCohortQueryReport(eq(CONFIG_NAME), eq(cohortQueryUuid), eq(null));
+
+        openMRSActionProxyService.getCohortQueryReport(CONFIG_NAME, cohortQueryUuid, null);
+
+        verify(cohortService).getCohortQueryReport(eq(CONFIG_NAME), uuidCaptor.capture(), parametersCaptor.capture());
+        verify(eventRelay, times(3)).sendEventMessage(Matchers.<MotechEvent>any());
+        assertEquals(cohortQueryUuid, uuidCaptor.getValue());
+        assertNull(parametersCaptor.getValue());
```
Comment (kmadej) [inline openmrs/src/test/java/org/motechproject/openmrs/tasks/impl/OpenMRSActionProxyServiceTest.java]: you should verify if sent parameters are empty

Comment (kmadej) [inline openmrs/src/test/java/org/motechproject/openmrs/tasks/impl/OpenMRSActionProxyServiceTest.java]: additionally, adjust test name

**Comment by martokarski** [comment]
test this please

**Pull request merged**